builtin/logical/aws/path_roles.go

```diff
@@ -222,6 +227,22 @@ func (b *backend) pathRolesWrite(ctx context.Context, req *logical.Request, d *f
 		roleEntry.DefaultSTSTTL = time.Duration(defaultSTSTTLRaw.(int)) * time.Second
 	}
 
+	if maxSTSTTLRaw, ok := d.GetOk("max_sts_ttl"); ok {
+		if legacyRole != "" {
+			return logical.ErrorResponse("cannot supply deprecated role or policy parameters with max_sts_ttl"), nil
+		}
+		if !strutil.StrListContains(roleEntry.CredentialTypes, assumedRoleCred) && !strutil.StrListContains(roleEntry.CredentialTypes, federationTokenCred) {
+			return logical.ErrorResponse(fmt.Sprintf("max_sts_ttl parameter only valid for %s and %s credential types", assumedRoleCred, federationTokenCred)), nil
+		}
+
+		maxSTSTTL := time.Duration(maxSTSTTLRaw.(int)) * time.Second
+		if roleEntry.DefaultSTSTTL > 0 && roleEntry.DefaultSTSTTL > maxSTSTTL {
```

Comment:
Two things:

First,m I think you should also test if `maxSTSTL > 0`. The reason is that, let's say a user has a `max_sts_ttl` of 3600 and a `default_sts_ttl` of 1200. Then, the user wants to remove the `max_sts_ttl`. That could be done by explicitly setting `max_sts_ttl` to 0, but then it would trip this check.

Second, you should probably move this check to after this if block. Imagine a user first sets `max_sts_ttl` to 1200, then in a second request, sets `default_sts_ttl` to 1800. That should probably also fail out, but I don't think it would.